catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -0,0 +1,245 @@
+
+  public CqlTransformHandler(
+      List<ServiceReference> queryResponseTransformers, BundleContext bundleContext) {
+    this.queryResponseTransformers = queryResponseTransformers;
+    this.bundleContext = bundleContext;
+  }
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    String transformerId = request.params(":transformerId");
+    String body = util.safeGetBody(request);
+    CqlRequest cqlRequest = mapper.readValue(body, CqlRequest.class);
+    CqlQueryResponse cqlQueryResponse = executeCqlQuery(cqlRequest);
+
+    LOGGER.trace("Finding transformer to transform query response.");
+
+    ServiceReference<QueryResponseTransformer> queryResponseTransformer =
+        queryResponseTransformers
+            .stream()
+            .filter(transformer -> transformer.getProperty("id").equals(transformerId))
+            .findFirst()
+            .orElse(null);
+
+    if (queryResponseTransformer == null) {
+      LOGGER.debug("Could not find transformer with id: {}", transformerId);
+      response.status(404);
+      return mapper.toJson(ImmutableMap.of("message", "Service not found"));
+    }
+
+    attachFileWithTransformer(queryResponseTransformer, cqlQueryResponse, request, response);
+
+    return "";
+  }
+
+  private void attachFileWithTransformer(
+      ServiceReference<QueryResponseTransformer> queryResponseTransformer,
+      CqlQueryResponse cqlQueryResponse,
+      Request request,
+      Response response)
+      throws CatalogTransformerException, MimeTypeException, IOException {
+
+    BinaryContent content;
+
+    content =
+        bundleContext
+            .getService(queryResponseTransformer)
+            .transform(cqlQueryResponse.getQueryResponse(), new HashMap<>());
+
+    String mimeType = (String) queryResponseTransformer.getProperty("mime-type");
+    MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+    String fileExt;
+
+    fileExt = allTypes.forName(mimeType).getExtension();
+
+    String acceptEncoding = request.headers(HttpHeaders.ACCEPT_ENCODING);
+
+    boolean shouldGzip =
+        StringUtils.isNotBlank(acceptEncoding) && acceptEncoding.toLowerCase().contains("gzip");
+
+    response.type(mimeType);
+    String attachment =
+        String.format("attachment;filename=export-%s%s", Instant.now().toString(), fileExt);
+    response.header(HttpHeaders.CONTENT_DISPOSITION, attachment);
+
+    try (OutputStream servletOutputStream = response.raw().getOutputStream();
+        InputStream resultStream = content.getInputStream()) {
+      if (shouldGzip) {
+        response.header(HttpHeaders.CONTENT_ENCODING, "gzip");
+        LOGGER.trace("Request header accepts gzip");
+        try (OutputStream gzipServletOutputStream = new GZIPOutputStream(servletOutputStream)) {
+          IOUtils.copy(resultStream, gzipServletOutputStream);
+        }
+      } else {
+        IOUtils.copy(resultStream, servletOutputStream);
+      }
+    }
+
+    response.status(200);
```

Comment:
✏️ use constant instead of 200 for status code.

Reply:
Honestly for status codes i don't see the need for abstracting it behind a variable. If someone then adds another 200 status code it will be easiest for them to just type 200 than to go find whatever someone has happened to name `200` in this class.